Tighten requirements, support Flask 2.3.x, track Fixture callbacks

crscheid opened the pull request:
This pull request addresses issue #49:

- Locks down the requirements to major versions via requirements.txt
- Supports named Blueprints with Flask 2.3 
- Increments version `2.0.0` to `2.0.1`

crscheid committed:
Tightened requirements, now supports Flask 2.3.x

crscheid commented:
Not sure why scrutinzer cannot find [pyftdi 0.55.0](https://pypi.org/project/pyftdi/0.55.0/) or [pyftdi 0.54.0](https://pypi.org/project/pyftdi/0.54.0/). Successfully tested on MacOS arm64 and Raspberry Pi platforms.

crscheid committed:
Modified _Fixture to support removal of local callbacks

crscheid committed:
Bumped version to 2.0.2

MattIPv4 commented on PyDMXControl/profiles/defaults/_Fixture.py:
```diff
@@ -56,6 +56,9 @@ def unpark(self):
 class FixtureHelpers:
 
+    # Store the callbacks
+    local_callback_list = []
```
Should this be `__local_callback_list` so that is it private? We could probably keep the naming simpler and just have it be `__callbacks`?

crscheid replied:
Good idea, will make that change.

MattIPv4 commented on PyDMXControl/profiles/defaults/_Fixture.py:
```diff
@@ -129,6 +136,12 @@ def locate(self):
         self.color([255, 255, 255, 255, 255])
         self.dim(255)
 
+    def remove_local_callbacks(self):
```
```suggestion
    def clear_callbacks(self):
```
Naming consistency with the effects method

MattIPv4 commented on PyDMXControl/profiles/defaults/_Fixture.py:
```diff
@@ -389,4 +402,7 @@ def clear_effects(self) -> 'Fixture':
         # Clear
         self.__effects = []
 
+        # Remove local callbacks - FixtureHelper
+        self.remove_local_callbacks()
```
I don't think this should be called here? Effects are tracked on their own in `__effects`, and that's what this method deals with. I think clearing callbacks that're outside effects should be its own call that a user needs to do?

crscheid replied:
Yeah I think that's a good idea. I'll make it a seperate call.

MattIPv4 commented:
> Not sure why scrutinzer cannot find [pyftdi 0.55.0](https://pypi.org/project/pyftdi/0.55.0/) or [pyftdi 0.54.0](https://pypi.org/project/pyftdi/0.54.0/). Successfully tested on MacOS arm64 and Raspberry Pi platforms.

It looks like scrutinzer is running Python 3.6, and 0.54.0 is set to Python 3.7+?

crscheid committed:
Update PyDMXControl/profiles/defaults/_Fixture.py

Naming consistency

Co-authored-by: Matt Cowley <me@mattcowley.co.uk>

crscheid commented:
I didn't realize that the pull request would update from my master after I committed the other changes for callbacks as I meant to do these two sets of changes separately.

That said, I will clean this up in my master branch of my fork and send a note when its done.

crscheid commented:
> > Not sure why scrutinzer cannot find [pyftdi 0.55.0](https://pypi.org/project/pyftdi/0.55.0/) or [pyftdi 0.54.0](https://pypi.org/project/pyftdi/0.54.0/). Successfully tested on MacOS arm64 and Raspberry Pi platforms.
> 
> It looks like scrutinzer is running Python 3.6, and 0.54.0 is set to Python 3.7+?

LMK if you plan to update to 3.7+ if not, I'll downgrade pyftdi to support Python 3.6+.

MattIPv4 commented:
3.6 has been EOL for a while now, so I'd be fine with bumping to 3.7 as part of this :)

crscheid committed:
Version bump to 2.1 since we added "clear_callbacks" to _Fixture. Revised internal tracking variable name. Removed clear_callbacks from Fixture class and left in FixtureHelper.

crscheid committed:
Bumped Python requirement to 3.7+

crscheid commented:
Happy Monday: changes to this PR from my master branch

- Updated setup to require Python >= 3.7 (scrutinizer config for this project might still require an update)
- Bumped version to `2.1.0` due to feature addition of `clear_callbacks` within FixtureHelper
- Modified internal tracking variable for fixtures to `__callbacks` as suggested
- Removed `remove_local_callbacks` from `clear_effects`. (Since `clear_callbacks` is in FixtureHelper, no need to have another function at the Fixture level since its already accessible.)

MattIPv4 commented on requirements.txt:
```diff
@@ -1,12 +1,12 @@
 # Web
-flask
\ No newline at end of file
+Flask>=2.3.3,<3.0
```
Is there reason to pin this at 2.3.3 specifically, rather than just 2.3?

crscheid replied:
Looking at the Flask Changelog, no not really so I changed to just 2.3. In other internal projects, I tend to use the latest working minor release version - just a habit I developed.

MattIPv4 reviewed (commented):
👀 Looks like Flask 2.3 requires Python 3.8. Given 3.7 is also EOL, this seems fine to bump to as well.

crscheid committed:
Bump required Python to 3.8 per Flask 2.3 requirement.

crscheid commented:
- Updated Python requirement to >=3.8
- Set Flask requirement at >=2.3

MattIPv4 reviewed (approved):
LGTM, tyvm! Sorry for the delay, this completely fell between the cracks in my inbox

crscheid commented:
All good, cheers!
